MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,37 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Don't display section header if there is only one step
+        guard routeController.routeProgress.route.legs.count > 1 else {
+            return nil
+        }
+        
+        let sectionName = routeController.routeProgress.route.legs[section].name
+        let legToFrom = sectionName.components(separatedBy: ", ")
+        
+        if legToFrom.count == 2 {
+            return "\(legToFrom[0]) to \(legToFrom[1])"
+        } else {
+            return sectionName
+        }
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return routeController.routeProgress.currentLeg.steps.count
+        return routeController.routeProgress.route.legs[section].steps.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: RouteTableViewCellIdentifier, for: indexPath) as! RouteTableViewCell
-        let leg = routeController.routeProgress.currentLeg
+        let legs = routeController.routeProgress.route.legs
+        
         
-        cell.step = leg.steps[indexPath.row]
+        cell.step = legs[indexPath.section].steps[indexPath.row]
         
-        if routeController.routeProgress.currentLegProgress.stepIndex + 1 > indexPath.row {
+        if routeController.routeProgress.legIndex + 1 > indexPath.section && routeController.routeProgress.currentLegProgress.stepIndex + 1 > indexPath.row  {
```

Comment:
We need to dim a cell for either of two reasons:

1. Either it represents a step that’s part of a leg that the user has already completed,
1. Or it represents a completed step in the current leg.

So the correct logic would be:

```swift
if indexPath.section < routeController.routeProgress.legIndex
    || (indexPath.section == routeProgress.legIndex
        && indexPath.row <= routeProgress.currentLegProgress.stepIndex) {
    // dim the cell
}
```